src/ui/PipelineContext/PipelineContext.ts

```diff
@@ -44,63 +46,97 @@ export interface IPipelineContextOptions {}
  * Using this component as opposed to JavaScript code means you will be able to leverage the interface editor.
  *
  * Regardless of if you use this component or JavaScript to add context, both will add the needed data in the [Query.Context]{@link IQuery.context} parameter.
+ * 
+ * **Note:**
+ * 
+ * Using this component also ensures that the framework will be able to properly determine the context in all corner cases, including when using a Standalone Search box ({@link initSearchbox}) displaying query suggestions.
+ * 
+ * If this component is not used in those cases, the context won't be able to be resolved and leveraged properly in the [Query Pipeline](http://www.coveo.com/go?dest=cloudhelp&lcid=9&context=108).
+ * 
  */
-export class PipelineContext extends Component {
+export class PipelineContext extends Component implements IPipelineContextProvider {
   static ID = 'PipelineContext';
   static CURRENT_URL = 'CurrentUrl';
 
   static doExport = () => {
     exportGlobally({
-      PipelineContext: PipelineContext,
-      context: context
+      PipelineContext: PipelineContext
     });
   };
 
-  private content: { [id: string]: string | Array<string> };
+  private contextContent: Context = {};
 
   public constructor(public element: HTMLElement, public options?: IPipelineContextOptions, public bindings?: IComponentBindings) {
     super(element, PipelineContext.ID, bindings);
     this.options = ComponentOptions.initComponentOptions(element, PipelineContext, options);
+    this.setContext(
+      $$(this.element)
+        .text()
+        .trim()
+    );
+    this.bind.onRootElement(QueryEvents.buildingQuery, (args: IBuildingQueryEventArgs) => this.handleBuildingQuery(args));
+  }
 
-    if (this.element.tagName.toLowerCase() == 'script') {
-      try {
-        // Content can be HTML encoded for special char ({!})
-        this.content = JSON.parse(Utils.decodeHTMLEntities($$(this.element).text()));
-      } catch (e) {
-        try {
-          this.content = JSON.parse($$(this.element).text());
-        } catch (e) {
-          return;
-        }
-      }
+  /**
+   * Set a new context, replacing any value previously set.
+   * 
+   * @param newContext The new context to set, which can be directly passed as a JSON, or as a stringified JSON.
+   */
+  public setContext(newContext: string | Context) {
+    if (_.isString(newContext)) {
+      const contextParsed = this.tryParseContextFromString(newContext);
+      this.contextContent = contextParsed;
+    } else {
+      this.contextContent = newContext;
     }
-    this.bind.onRootElement(QueryEvents.buildingQuery, (args: IBuildingQueryEventArgs) => this.handleBuildingQuery(args));
+  }
+
+  /**
+   * Return the current context
```

Comment:
Suggestion:

> Returns the current context.

Reply:
I've opened a doc issue to review and uniformize method documentation (see DOC-1715).